### DISCUSSION TITLE
Conditionals in lists

### DISCUSSION THREAD

**Discussion by eonpatapon** [cuelang/cue#713] [Q&A] [answered]
It is now possible to include multiple comprehensions and `if` conditions in lists:

```
a: true

l: [
	if a {
		"foo"
	},
	"foobar",
]
```

Returns:
```
a: true
l: ["foo", "foobar"]
```

However it doesn't seem possible to include multiple items in the `if` clause. This doesn't work:

```
a: true

l: [
	if a {
		"foo",
		"bar",
	},
	"foobar",
]
```

Returns:
```
 cue eval x.cue
l.0: conflicting values "bar" and "foo":
```

Is there a workaround or plans to make this work ?

**Comment by myitcv** [comment, marked as answer]
I'll leave @mpvl to comment on whether anything like this would ever make sense.

But for now you can:

```
package x

a: true

l: [
	for _, v in ["foo", "bar"] if a {
		v
	},
	"foobar",
]
```

**Reply by mpvl** [reply]
This is the correct answer. The other thing indeed would never make sense, as `{ "foo", "bar" }` inherently means `{ "foo" & "bar" }` a per the embedding rules (and it would be weird to make an exception in this case).

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/713.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
